Add in support for relative timestamps.

mwringe opened the pull request:
See https://issues.jboss.org/browse/HWKMETRICS-358

burmanm commented:
I think this feature should use Durations for these timestamps (see org.hawkular.metrics.model.param.Duration)

mwringe commented:
Updated to use durations

mwringe commented:
retest this please

stefannegrea commented on core/metrics-model/src/main/java/org/hawkular/metrics/model/param/TimeRange.java:
```diff
@@ -31,24 +31,52 @@
+        if (number.startsWith("+")) {
+            String offset = number.substring(1);
+            Duration duration = new Duration(offset);
+            return now + duration.getValue();
```
I do not think this is correct. I think you need to get the value in millis of the duration.

stefannegrea commented on core/metrics-model/src/main/java/org/hawkular/metrics/model/param/TimeRange.java:
```diff
@@ -31,24 +31,52 @@
+    private Long handleOffsets(String number) throws NumberFormatException, IllegalArgumentException{
+        long now = System.currentTimeMillis();
+        if (number.startsWith("+")) {
+            String offset = number.substring(1);
```
I would like to see some trimming of white spaces after removing the + or - because the users could send something like `+ 10s`.

stefannegrea commented on core/metrics-model/src/test/java/org/hawkular/metrics/model/param/TimeRangeTest.java:
```diff
@@ -50,20 +50,84 @@ public void testLowerBoundDefault() {
     }
 
     @Test
+    public void testRelativeTimeStamps() {
```
Do you mind adding some tests with deltas with time units other than ms?

mwringe commented:
Updated to make sure we are receiving the right millisecond values from the duration.
Updated to trim the space after "+" or "-"
Updated tests to check different duration values other than just ms

stefannegrea commented:
Thanks for the updates @mwringe. I will merge as soon as the perf tests pass.
